### PR TITLE
Set NIX_SSL_CERT_FILE in the daemon

### PR DESCRIPTION
**Pull request by Hoverbear**
##### Description

Investigating #289 more, I asked @edolstra for some guidance around when to set `NIX_SSL_CERT_FILE`. Resulting from that discussion, this PR adds the necessary environment settings into the init services of both Mac and Linux.

A side note: It would be good to split up `configure_init_service.rs` more...

##### Checklist

- [ ] Formatted with `cargo fmt`
- [ ] Built with `nix build`
- [ ] Ran flake checks with `nix flake check`
- [ ] Added or updated relevant tests (leave unchecked if not applicable)
- [ ] Added or updated relevant documentation (leave unchecked if not applicable)
- [ ] Linked to related issues (leave unchecked if not applicable)

##### Validating with `install.determinate.systems`

If a maintainer has added the `upload to s3` label to this PR, it will become available for installation via `install.determinate.systems`:

```shell
curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix/pr/$PR_NUMBER | sh -s -- install
```

### CODE REVIEW

**Comment by Hoverbear** [comment]
image

Demonstrating that even without a `ca-certificates.crt` on system this works:

image

File: src/action/common/configure_init_service.rs [hunk 1]
```diff
@@ -177,6 +198,18 @@ impl Action for ConfigureInitService {
                 )
                 .await?;
 
+                if let Some(ssl_cert_file) = ssl_cert_file {
+                    execute_command(
+                        Command::new("launchctl")
+                            .process_group(0)
+                            .arg("setenv")
+                            .arg("NIX_SSL_CERT_FILE")
+                            .arg(format!("{ssl_cert_file:?}"))
```
Comment (cole-h) [inline src/action/common/configure_init_service.rs:207]: Why debug format?

File: src/action/common/configure_init_service.rs [hunk 2]
```diff
@@ -270,6 +303,28 @@ impl Action for ConfigureInitService {
...
+                            ActionError::CreateDirectory(service_conf_dir_path.clone(), e)
+                        })?;
+                    let service_conf_file_path =
+                        service_conf_dir_path.join("nix-ssl-cert-file.conf");
+                    tokio::fs::write(
+                        service_conf_file_path,
+                        format!(
+                            "\
+                        [Service]\n\
+                        Environment=\"NIX_SSL_CERT_FILE={ssl_cert_file:?}\"\n\
```
Comment (cole-h) [inline src/action/common/configure_init_service.rs:320]: Same question.

File: src/action/common/configure_init_service.rs (hunk 1 above)
Comment (Hoverbear) [reply src/action/common/configure_init_service.rs:207]: Great question!

> If you would like an implementation which escapes the path please use [Debug](https://doc.rust-lang.org/std/fmt/macro.Debug.html) instead.
> - https://doc.rust-lang.org/std/path/struct.PathBuf.html#method.display

File: src/action/common/configure_init_service.rs (hunk 2 above)
Comment (Hoverbear) [reply src/action/common/configure_init_service.rs:320]: Great same question!

> If you would like an implementation which escapes the path please use [Debug](https://doc.rust-lang.org/std/fmt/macro.Debug.html) instead.
> - https://doc.rust-lang.org/std/path/struct.PathBuf.html#method.display

**Review by cole-h — APPROVED** [review]

**Comment by Hoverbear** [comment]
Confirmed this is working on both Linux and Mac manually. I'd like to figure out a test eventually but that can wait.

**Comment by Hoverbear** [comment]
Oh shoot I accidently merged a branch here...

**Pull request merged**